### ISSUE TITLE
Uninitialized constant Azure::Core::Http::HttpRequest::Tempfile (NameError)

### ISSUE THREAD

**Issue by kac460** [Azure/azure-storage-ruby#162]
Following https://docs.microsoft.com/en-us/azure/storage/blobs/storage-quickstart-blobs-ruby results in `Uninitialized constant Azure::Core::Http::HttpRequest::Tempfile (NameError)`. I used version 2.0.0 of the gem. I was using Ruby 2.6.3 when I encountered this. I also tried Ruby 2.7.0 and got the same error. The location of the problem is here: https://github.com/Azure/azure-storage-ruby/blob/e5059568b6b971f6981b454fd71cf211429d418e/common/lib/azure/core/http/http_request.rb#L162

For what it's worth, I found that a possible solution is to add `require 'tempfile'` to https://github.com/Azure/azure-storage-ruby/blob/e5059568b6b971f6981b454fd71cf211429d418e/common/lib/azure/core/http/http_request.rb .

**Comment by yaxia** [comment]
Hi @kac460,
Thanks for reporting the issue. However I am not able to reproduce this issue using Ruby 2.7.

Could you please folk this repo and run below command in the root folder of this repo:
```` ruby
ruby -I"common/lib;test" test/unit/core/http/http_request_test.rb
````

This the test file for verifying functions of the `http_request.rb` and it doesn't need explicit `require 'tempfile'`. Please let me know what you get from the output. Thanks.

**Comment by pawel-f** [comment]
hi @yaxia , 

I confirm that I have encountered the same issue. I tested it with Ruby 2.3 and 2.5: 
`.rvm/gems/ruby-2.3.4/gems/azure-storage-common-2.0.1/lib/azure/core/http/http_request.rb:163:in "apply_body_headers': uninitialized constant Azure::Core::Http::HttpRequest::Tempfile (NameError)"`

Adding explicit `require 'tempfile'` to the `http_request.rb` solves the issue.
I'm using a local blob storage from Azure IoT Edge on K8s.

**Comment by yaxia** [comment]
> hi @yaxia ,
> 
> I confirm that I have encountered the same issue. I tested it with Ruby 2.3 and 2.5:
> `.rvm/gems/ruby-2.3.4/gems/azure-storage-common-2.0.1/lib/azure/core/http/http_request.rb:163:in "apply_body_headers': uninitialized constant Azure::Core::Http::HttpRequest::Tempfile (NameError)"`
> 
> Adding explicit `require 'tempfile'` to the `http_request.rb` solves the issue.
> I'm using a local blob storage from Azure IoT Edge on K8s.

Thanks for the update. I will take a look. Please stay tuned.

**Comment by jaroslawkozak** [comment]
Hello,
Any updates on this issue? `require 'tempfile'` solution works, but I'd rather avoid messing up with gems, especially when dockerizing ruby.

**Comment by katmsft** [comment]
This is fixed in 2.0.1/2.0.2 release, check it out!
